table/tables/tables.go

```diff
@@ -509,13 +511,31 @@ func (t *tableCommon) AddRecord(ctx sessionctx.Context, r []types.Datum, opts ..
 		if !t.canSkip(col, value) {
 			colIDs = append(colIDs, col.ID)
 			row = append(row, value)
+			continue
+		}
+		// Fill column size for statistics.
+		if col.IsPKHandleColumn(t.Meta()) {
+			// The size of primary key column is 8 byte.
+			colSize[col.ID] = 8
+			continue
+		}
+		if col.IsGenerated() && !col.GeneratedStored {
+			// Lazy-initialization buffer.
+			if encodedCol == nil {
+				encodedCol = make([]byte, 0, 16)
```

Comment:
why 16 here?

Reply:
The original value is 16, I don't change it.